From 62fec746653a907ad6569788c26bfcf6f359c4d6 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Tue, 4 Aug 2020 11:53:02 +0530
Subject: [PATCH 21/21] Our PWM backlight has an enable pin

This patch adds LCD_BKLT_EN as enable GPIO to the existing backlight node.
Our LCD_BKLT_PWM is also muxed differently than SabreSD's. Default
brightness is set to 50%, that way the panel will be lit regardless of PWM
polarity.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 02bdaea82200..2b57a517129c 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -269,7 +269,8 @@
 		compatible = "pwm-backlight";
 		pwms = <&pwm1 0 5000000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
-		default-brightness-level = <7>;
+		default-brightness-level = <6>;
+		enable-gpios = <&gpio6 10 GPIO_ACTIVE_HIGH>; /* SMARC LCD_BKLT_EN */
 		status = "okay";
 	};
 
@@ -806,7 +807,6 @@
 				MX6QDL_PAD_KEY_ROW2__GPIO4_IO11	0x80000000 /* SMARC SDIO_PWR_EN */
 				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x80000000
 				MX6QDL_PAD_SD3_RST__GPIO7_IO08 0x80000000
-				MX6QDL_PAD_GPIO_9__GPIO1_IO09  0x80000000
 				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
 			>;
 		};
@@ -1068,7 +1068,8 @@
 
 		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
-				MX6QDL_PAD_GPIO_9__PWM1_OUT	0x1b0b1 /* SMARC LCD_BKLT_PWM */
+				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10	0x80000000 /* SMARC LCD_BKLT_EN */
+				MX6QDL_PAD_GPIO_9__PWM1_OUT		0x1b0b1 /* SMARC LCD_BKLT_PWM */
 			>;
 		};
 
@@ -1218,7 +1219,7 @@
 	status = "okay";
 };
 
-&pwm1 {
+&pwm1 {	/* SMARC LCD_BKLT_PWM */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pwm1>;
 	status = "okay";
-- 
2.17.1

